### PR TITLE
Remove curl and parallelize node/bun install

### PR DESCRIPTION
**Pull request by picklelo**
Use `httpx` to install nvm and bun instead of curl, removing one dependency. Also parallelize the process to install nvm/bun during `reflex init`

### CODE REVIEW

**Comment by jackie-pc** [comment]
Possible to base this PR on the other "package.json" PR, so we see what changed?

File: reflex/utils/prerequisites.py [hunk 1]
```diff
@@ -319,27 +336,35 @@ def install_bun():
...
-            raise FileNotFoundError("Reflex requires unzip to be installed.")
-
-        result = subprocess.run(constants.INSTALL_BUN, shell=True)
+    # Skip if bun is already installed.
+    if os.path.exists(constants.BUN_PATH):
+        return
 
-        if result.returncode != 0:
-            raise typer.Exit(code=result.returncode)
+    # Check if unzip is installed
```
Comment (jackie-pc) [inline reflex/utils/prerequisites.py:345]: We should create an issue to remove unzip as a requirement as well.

File: reflex/utils/prerequisites.py [hunk 2]
```diff
@@ -319,27 +336,35 @@ def install_bun():
...
+    unzip_path = path_ops.which("unzip")
+    if unzip_path is None:
+        raise FileNotFoundError("Reflex requires unzip to be installed.")
+
+    # Get the bun install script.
+    response = httpx.get(constants.BUN_INSTALL_URL)
```
Comment (jackie-pc) [inline reflex/utils/prerequisites.py:351]: We can potentially dedupe this further.... `def download_and_run()`:  same for both nvm and bun.

File: reflex/utils/prerequisites.py [hunk 3]
```diff
@@ -281,31 +284,45 @@ def install_node():
...
-    console.log("Installing node...")
-    result = subprocess.run(constants.INSTALL_NODE, shell=True)
-
+    # Install node.
+    print("installing node")
+    result = subprocess.run(
+        [
+            "bash",
+            "-c",
+            f". {constants.NVM_DIR}/nvm.sh && nvm install {constants.NODE_VERSION}",
```
Comment (jackie-pc) [inline reflex/utils/prerequisites.py:322]: Add comment that we have to do this "bash -c" business because we need to source in `. nvm.sh`?

**Comment by jackie-pc** [comment]
We can verify curl is gone as a dependency by doing an `apt remove` [here](https://github.com/reflex-dev/reflex/blob/afb15109db0f0e3b040a88b5a37f49b00f6dd699/integration/init-test/Dockerfile#L18C5-L18C5).  Or we can just not use curl (e.g. by using wget instead, etc) within the Dockerfile itself.

It's fine to also file a new issue for this part IMO.  Benefit is mainly to ensure we catch any future addition of this dependency and that there are no other indirect deps on curl (e.g. within nvm.sh, etc)

File: reflex/utils/prerequisites.py (hunk 1 above)
Comment (picklelo) [reply reflex/utils/prerequisites.py:357]: Agreed, created this ticket #1459

File: reflex/utils/prerequisites.py (hunk 3 above)
Comment (picklelo) [reply reflex/utils/prerequisites.py:334]: added a comment

File: reflex/utils/prerequisites.py (hunk 2 above)
Comment (picklelo) [reply reflex/utils/prerequisites.py]: Great idea, factored this out to a function

**Review by jackie-pc — DISMISSED** [review]

**Review by Alek99 — APPROVED** [review]

**Pull request merged**